---
title: "TK1 Expression in HER2+ vs. TNBC"
output: html_document
---
### Parsing Data  
All datasets are based on [The Cancer Genome Atlas](https://cancergenome.nih.gov/) (TCGA). The specific version of the data used for analysis can be found [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62944). 

```{r setup, message=FALSE, warning=FALSE}
library(rmarkdown)
library(dplyr)
library(ggplot2)
library(tidyverse)
require(readr)
```
  
The file below contains a list of patient IDs in TCGA with various cancer types. About 1,000 of these patients had breast cancer (indicated by "BRCA" in the second column). IDs with breast cancer are extracted from this file.  
```{r, message=FALSE, warning=FALSE}
# Download and read in file
download.file('https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE62944&format=file&file=GSE62944%5F06%5F01%5F15%5FTCGA%5F24%5FCancerType%5FSamples%2Etxt%2Egz', 
              destfile = "./GSE62944_06_01_15_TCGA_24_CancerType_Samples.txt.gz")
brca_IDs <- as.data.frame(read_tsv("GSE62944_06_01_15_TCGA_24_CancerType_Samples.txt.gz"))

colnames(brca_IDs) <- c("Patient", "CancerType")

# Extract patient IDs with BRCA
brca_IDs <- brca_IDs[grep("BRCA", brca_IDs$Cancer), ]
```
  
The file below contains clinical data about TCGA patients from the first file we parsed. Data is parsed for BRCA IDs only and gathered from rows that start with the following: 

* er_status_by_ihc  
* her2_status_by_ihc  
* pr_status_by_ihc  

If the patient's status is negative in all three categories, then the patient has TNBC.  
If the patient's status is positive in her2_status_by_ihc, then the patient is HER2+.
```{r, message=FALSE, warning=FALSE}
# Download and read in file
download.file('https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE62944&format=file&file=GSE62944%5F06%5F01%5F15%5FTCGA%5F24%5F548%5FClinical%5FVariables%5F9264%5FSamples%2Etxt%2Egz', 
              destfile = "./GSE62944_06_01_15_TCGA_24_548_Clinical_Variables_9264_Samples.txt.gz")
clinical_data <- as.data.frame(read_tsv('GSE62944_06_01_15_TCGA_24_548_Clinical_Variables_9264_Samples.txt.gz'))

rownames(clinical_data) <- clinical_data$X1
clinical_data <- clinical_data[-c(1:3)]

# Extract BRCA patient IDs only (cols) and ER, HER2, PR (rows)
clinical_data <- clinical_data[, names(clinical_data)[names(clinical_data) %in% brca_IDs$Patient]]
clinical_data <- clinical_data[c("er_status_by_ihc", "her2_status_by_ihc", "pr_status_by_ihc"),]

# Transpose data frame
clinical_data <- t(clinical_data)
clinical_data <- cbind(rownames(clinical_data), data.frame(clinical_data, row.names = NULL))
colnames(clinical_data) <- c("Patient", "ER", "HER2", "PR")

# If not "Positve" and "Negative" change info to NA and remove rows that have NA in any column
clinical_data$ER[(clinical_data$ER != "Positive" & clinical_data$ER != "Negative")] <- NA
clinical_data$HER2[(clinical_data$HER2 != "Positive" & clinical_data$HER2 != "Negative")] <- NA
clinical_data$PR[(clinical_data$PR != "Positive" & clinical_data$PR != "Negative")] <- NA
clinical_data <- clinical_data[complete.cases(clinical_data),]

# TNBC: negative in ER, HER2, PR
clinical_data$TNBC <- ifelse(
  clinical_data$ER == "Negative" & clinical_data$HER2 == "Negative" & clinical_data$PR == "Negative", TRUE, FALSE) 

# Status: TNBC, HER2+, Other
clinical_data$Status <- ifelse(clinical_data$TNBC==TRUE, "TNBC", 
                               ifelse(clinical_data$HER2=="Positive", "HER2+", "Other"))
clinical_data <- filter(clinical_data, Status != "Other")
clinical_data$Status <- factor(clinical_data$Status, levels=c("HER2+", "TNBC"))

# Keep only the first 12 characters of IDs (to merge with RNA data later on)
clinical_data$Patient <- substr(clinical_data$Patient, 0, 12)
```
  
The file containing gene expression (RNA sequencing) data is then parsed and merged with the clinical data.
```{r, message=FALSE, warning=FALSE}
# Read in file
rna_data <- as.data.frame(read_tsv('Evita_RNASeq.tsv.gz'))
colnames(rna_data)[1] <- "Patient"

# Combine both data frames by patient IDs
total_data <- merge(clinical_data, rna_data, by = "Patient")
total_data$CancerType <- NULL
```
  
  
### Plots
The merged data is prepped in order to create a boxplot and several scatterplots.  
The gene expression values are log2-transformed in order to mitigate the effects of outliers. 1 is also added to prevent errors in case there are any zero values. 
```{r, message=FALSE, warning=FALSE}
# log2-transform expression values
stemness_genes <- names(total_data)[names(total_data) %in% 
                                      c("TK1", "CD44", "SNAI1", "SNAI2", "TWIST1", "ZEB1", "TGFB1")]

for (gene_col in stemness_genes) {
  total_data[, gene_col] <- log2(total_data[,gene_col] + 1)
}

# Filter data by status
HER2 <- filter(total_data, Status == "HER2+")
TNBC <- filter(total_data, Status == "TNBC")
```
  
The boxplot outputted by the code below illustrates TK1 expression in HER2+ and TNBC samples.  
Statistical analysis is performed using the [Wilcoxon test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test), which ranks data to determine if a significant difference exists between the samples. The p-value is provided in the title of the boxplot.
```{r, message=FALSE, warning=FALSE}
# Wilcoxon test
wilcox <- wilcox.test(HER2$TK1, TNBC$TK1)
wilcox_p <- formatC(wilcox$p.value, format = "g", digits = 2)

# Create boxplot
ggplot(total_data, aes(x=Status, y=TK1)) + 
  geom_boxplot(coef = 100, outlier.shape = NA) + 
  geom_jitter(alpha = 0.3) +
  ylab("TK1 Expression\n(TPM, log2 transformed)") +
  ggtitle(paste("TK1 Expression by Subtype (p=", wilcox_p, ")", sep="")) +
  theme_bw(base_size = 18) +
  theme(plot.title = element_text(hjust = 0.5))

# Save boxplot as a pdf file
ggsave(file="TCGA_boxplot.pdf", width=6.5)
```
  
  
The scatterplots outputted by the code below illustrate the correlation of expression between TK1 and each "stemness" gene for HER2+ and TNBC samples individually. The [Spearman correlation coefficient](https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient) (Ï or "rho") is calculated and included in the title of each scatterplot to statistically display the correlation. 
```{r, message=FALSE, warning=FALSE}
make_scatterplot <- function(df, stem_gene, title) {
  # Spearman correlation coefficient
  rho <- cor(df$TK1, df[stem_gene], method="spearman")
  rho <- round(rho, digits=2)
  # Create scatterplots
  ggplot(df, aes(x = TK1, y = df[stem_gene])) + 
    geom_point() + 
    geom_smooth(method='lm', color="red", se=FALSE) +
    ylab(stem_gene) + 
    ggtitle(paste(title, ": TK1 vs. ", stem_gene, " (rho=", rho, ")", sep="")) +
    scale_x_continuous() + 
    scale_y_continuous() + 
    theme_bw(base_size = 18) +
    theme(plot.title = element_text(hjust = 0.5))
}

# Save each scatterplot as a pdf file
save_pdf <- function(subtype, gene) {
  file_name <- paste("TCGA_scatterplot_", subtype, "_", gene, ".pdf", sep="")
  ggsave(file_name, width=6.5)
}

for (gene in stemness_genes[2:7]) {
  print(make_scatterplot(HER2, gene, "HER2+"))
  save_pdf("HER2", gene)
  print(make_scatterplot(TNBC, gene, "TNBC"))
  save_pdf("TNBC", gene)
}

```

